Content.Goobstation.Shared/MartialArts/SharedMartialArtsSystem.PlasmaFist.cs

```diff
@@ -0,0 +1,178 @@
+using Content.Shared.Body.Systems;
+using Content.Shared.Body.Components;
+using Robust.Shared.Physics.Components;
+using Content.Shared.Atmos.Components;
+using Robust.Shared.Physics.Systems;
+
+namespace Content.Goobstation.Shared.MartialArts;
+
+public partial class SharedMartialArtsSystem
+{
+    [Dependency] private readonly SharedBodySystem _bodySystem = default!;
+    [Dependency] private readonly EntityManager _entityManager = default!;
+
+    private void InitializePlasmaFist()
+    {
+        SubscribeLocalEvent<CanPerformComboComponent, TornadoSweepPerformedEvent>(OnTornadoSweep);
+        SubscribeLocalEvent<CanPerformComboComponent, ThrowbackPerformedEvent>(OnThrowback);
+        SubscribeLocalEvent<CanPerformComboComponent, ThePlasmaFistPerformedEvent>(OnThePlasmaFist);
+
+        SubscribeLocalEvent<GrantPlasmaFistComponent, UseInHandEvent>(OnGrantCQCUse);
+        SubscribeLocalEvent<GrantPlasmaFistComponent, ExaminedEvent>(OnGrantCQCExamine);
+
+    }
+
+    #region Generic Methods
+
+    private void OnGrantCQCUse(EntityUid ent, GrantPlasmaFistComponent comp, UseInHandEvent args)
+    {
+        if (!_netManager.IsServer)
+            return;
+
+        if (comp.Used)
+        {
+            _popupSystem.PopupEntity(Loc.GetString(comp.LearnFailMessage, ("manual", Identity.Entity(ent, EntityManager))),
+                args.User,
+                args.User);
+            return;
+        }
+
+        if (!TryGrantMartialArt(args.User, comp))
+            return;
+        _popupSystem.PopupEntity(Loc.GetString(comp.LearnMessage), args.User, args.User);
+        comp.Used = true;
+    }
+    #endregion
+
+    #region Combo Methods
+
+    private void OnTornadoSweep(Entity<CanPerformComboComponent> ent, ref TornadoSweepPerformedEvent args)
+    {
+        if (!_proto.TryIndex(ent.Comp.BeingPerformed, out var proto)
+            || !TryUseMartialArt(ent, proto, out var target, out var downed))
+            return;
+
+        var mapPos = _transform.GetMapCoordinates(ent).Position;
+        var hitPos = _transform.GetMapCoordinates(target).Position;
+        var dir = hitPos - mapPos;
+        dir *= 1f / dir.Length();
+
+        if (TryComp<PullableComponent>(target, out var pullable))
+            _pulling.TryStopPull(target, pullable, ent, true);
+
+        var xform = _entityManager.GetComponent<TransformComponent>(ent);
+        var mapId = xform.MapID;
+        var position = _transform.GetMapCoordinates(ent).Position;
+
+        var physicsQuery = _entityManager.GetEntityQuery<PhysicsComponent>();
+        var movedByPressureQuery = _entityManager.GetEntityQuery<MovedByPressureComponent>();
+
+        // Use the entity lookup system to find entities within the radius.
+        // Iterate over all entities with PhysicsComponent and TransformComponent
+        var entities = new List<EntityUid>();
+        var query = _entityManager.AllEntityQueryEnumerator<PhysicsComponent, TransformComponent>();
```

Comment:
Use LookupSystem to get entities in range instead of querying all physics entities.